resolver/GlobalPass.cc

```diff
@@ -360,7 +360,24 @@ void Resolver::computeLinearization(core::GlobalState &gs) {
     for (int i = 1; i < gs.classAndModulesUsed(); ++i) {
         const auto &ref = core::SymbolRef(&gs, core::SymbolRef::Kind::ClassOrModule, i);
         ENFORCE(ref.data(gs)->isClassOrModule());
-        computeClassLinearization(gs, ref);
+        auto mixins = computeClassLinearization(gs, ref).mixins;
+
+        // Iterate over mixins of the class as long as they are calling `mixes_in_class_methods()`
+        // Provides recursive `mixes_in_class_methods` support
+        core::SymbolRef singleton;
+        for (auto mod : mixins) {
+            auto mixedInClassMethod = mod.data(gs)->findMember(gs, core::Names::classMethods());
+            if (!mixedInClassMethod.exists()) {
+                break;
+            }
+            if (!singleton.exists()) {
+                singleton = ref.data(gs)->singletonClass(gs);
+            }
+
+            if (!singleton.data(gs)->addMixin(gs, mixedInClassMethod)) {
+                ENFORCE(false);
```

Comment:
Method definition mentions that caller should error in this case. I followed https://github.com/sorbet/sorbet/blob/d0119cbdcba3771b493e677f02703481de5f1077/resolver/GlobalPass.cc#L387-L391 and used `ENFORCE`.